src/common/main.lua

```diff
@@ -69,11 +70,87 @@ main.init = function()
 			["Message-Id"] = msg.Id,
 		})
 		-- Send HyperBEAM patch message with updated ACL that has the ant removed
+		-- only send if the acl is not empty, as we will nuke the entire ACL from the patch device if it is empty
+		if #utils.keys(acl) > 0 then
+			ao.send({
+				device = "patch@1.0",
+				cache = { acl = acl },
+			})
+		end
+	end)
 
-		ao.send({
-			device = "patch@1.0",
-			cache = { acl = acl },
-		})
+	utils.createActionHandler(ActionMap.BatchUnregister, function(msg)
+		assert(msg.From == Owner, "Only ANT Registry owner can batch unregister")
+		local antIds = json.decode(msg.Data)
```

Comment:
i forget, what does json.decode if msg.Data is an invalid json string?

Reply:
it blows up